homeassistant/components/climate/coolmaster.py

```diff
@@ -0,0 +1,205 @@
+        self._current_temperature = status['temperature']
+        self._current_fan_mode = status['fan_speed']
+        self._on = status['is_on']
+
+        device_mode = status['mode']
+        if device_mode == FAN_ONLY_OVERRIDE:
+            self._current_operation = STATE_FAN_ONLY
+        else:
+            self._current_operation = device_mode
+
+        if status['unit'] == 'celsius':
+            self._unit = TEMP_CELSIUS
+        else:
+            self._unit = TEMP_FAHRENHEIT
+
+    @property
+    def unique_id(self):
+        """Return unique ID for this device."""
+        return self._uid
+
+    @property
+    def device_info(self):
+        """Return information about the device."""
+        return {
+            'identifiers': {
+                (DOMAIN, self.unique_id),
+            },
+            'name': self.name,
+            'manufacturer': 'CoolMasterNet',
+        }
+
+    @property
+    def supported_features(self):
+        """Return the list of supported features."""
+        return SUPPORT_FLAGS
+
+    @property
+    def should_poll(self):
+        """Return the polling state."""
+        return True
+
+    @property
+    def name(self):
+        """Return the name of the climate device."""
+        return self.unique_id
+
+    @property
+    def temperature_unit(self):
+        """Return the unit of measurement."""
+        return self._unit
+
+    @property
+    def current_temperature(self):
+        """Return the current temperature."""
+        return self._current_temperature
+
+    @property
+    def target_temperature(self):
+        """Return the temperature we are trying to reach."""
+        return self._target_temperature
+
+    @property
+    def current_operation(self):
+        """Return current operation ie. heat, cool, idle."""
+        return self._current_operation
+
+    @property
+    def operation_list(self):
+        """Return the list of available operation modes."""
+        return self._operation_list
+
+    @property
+    def is_on(self):
+        """Return true if the device is on."""
+        return self._on
+
+    @property
+    def current_fan_mode(self):
+        """Return the fan setting."""
+        return self._current_fan_mode
+
+    @property
+    def fan_list(self):
+        """Return the list of available fan modes."""
+        return FAN_MODES
+
+    def set_temperature(self, **kwargs):
+        """Set new target temperatures."""
+        temp = kwargs.get(ATTR_TEMPERATURE)
+        if temp is not None:
+            _LOGGER.debug("Setting temp of %s to %s", self.unique_id,
+                          str(temp))
+            self._device.set_thermostat(str(temp))
+            self.schedule_update_ha_state()
+
+    def set_fan_mode(self, fan_mode):
+        """Set new fan mode."""
+        _LOGGER.debug("Setting fan mode of %s to %s", self.unique_id,
+                      fan_mode)
+        self._device.set_fan_speed(fan_mode)
+        self.schedule_update_ha_state()
```

Comment:
We can remove this. It will be called at the end of the service call automatically, since this is a polling entity.